lib/cosmos/io/json_drb.rb

```diff
@@ -46,52 +66,44 @@ def initialize
       @request_times = []
       @request_times_index = 0
       @request_mutex = Mutex.new
-      @client_sockets = []
-      @client_threads = []
-      @client_pipe_writers = []
-      @client_mutex = Mutex.new
-      @thread_reader, @thread_writer = IO.pipe
+      @server = nil
+      @server_mutex = Mutex.new
     end
 
     # Returns the number of connected clients
     # @return [Integer] The number of connected clients
     def num_clients
-      @client_threads.length
+      clients = 0
+      @server_mutex.synchronize do
+        if @server
+          # @server.stats() returns a string like: { "backlog": 0, "running": 0 }
+          # "running" indicates the number of server threads running, and
+          # therefore the number of clients connected.
+          stats = @server.stats()
+          stats =~ /"running": \d*/
+          clients = $&.split(":")[1].to_i
+        end
+      end
+      return clients
     end
 
     # Stops the DRb service by closing the socket and the processing thread
     def stop_service
-      Cosmos.kill_thread(self, @thread)
+      Cosmos.kill_thread(self, @thread, 5, 0.1, 5)
```

Comment:
Can I get a comment about these parameters to kill_thread?

Reply:
Done.